torchao/sparsity/prototype/training/README.md

```diff
@@ -0,0 +1,53 @@
+# Accelerated Sparse Training
+
+This folder contains a prototype implementation of fast sparse training, utilizing the runtime semi-structured (2:4) sparsification routines present in core.
+
+### Quickstart
+**NOTE: This feature is only available on the pytorch / torchao nightlies currently. **
+
+```python
+import torch
+from torchao.sparsity.prototype.training import (
+    SemiSparseLinear,
+    SemiSparseActivationLinear,
+    swap_linear_with_semi_sparse_linear_,
+    swap_semi_sparse_linear_with_linear_,
+)
+
+model = torch.nn.Sequential(torch.nn.Linear(64, 64)).cuda().to(torch.bfloat16)
```

Comment:
does the example actually run faster? If not try to to have the minimal example that'll run faster and a way of printing that speedup to console. Also specify the supported SM for those speedups since if it's ampere+ then 3090 and 4090 should also benefit from your work

Reply:
Updated, the example will run faster, and added compute capability limitations earlier in the readme, but for printing speedups - I don't think it's necessary as we have the benchmark script for that. 


modifying the benchmarks for the example I see the following speedup (fw pass only):
```
[------------------------------------------------ mlpfw -------------------------------------------------]  
                                  |  act24   |  dense   |   w24    |  s24_inp_sparsify24  |  s24_inp_clone
1 threads: -----------------------------------------------------------------------------------------------
      f16 (44160,1024,4096,1024)  |  4813.2  |  4031.0  |  3440.1  |        255.4         |      121.4    
```